lib/bot.js

```diff
@@ -1,221 +1,25 @@
 // @flow
-import getKeybaseUsernameAndDevicename from './keybase-status.js'
-import {runApiCommand} from './chat-api.js'
-import keybaseExec from './keybase-exec'
-import type {ApiCommandArg} from './chat-api.js'
-import type {
-  InitOptions,
-  DeviceUsernamePair,
-  ChatChannel,
-  ChatOptionsList,
-  ChatOptionsSend,
-  ChatReadMessage,
-  ChatOptionsDelete,
-  ChatOptionsRead,
-} from './types.js'
+import Base from './base'
+import Chat from './chat'
+import {createStore, reducer, initialState} from './store'
+import type {Store} from './store/types'
 
-/**
- * A Keybase bot.
- *
- * @class Bot
- */
 class Bot {
-  _dPair: ?DeviceUsernamePair
-  _initialized: boolean
-  _verbose: boolean
+  store: Store
+  init: any
```

Comment:
Is it possible to enforce types here? For example, I think `init` should have a type of `InitOptions => Promise<void>`, at least according to the docs.

I get that `withStore` kinda adds a lot of complexity to the typing, though, and I'm sure you've looked into this.

Reply:
@nathunsmitty Definitely on the same page here. I was wrangling types for a while as I was writing `withStore`. It's tough to get Flow to recognize different possible types in a `Union` type when each one is a function definition.

I might go back to the drawing board on this soon. I'll make a separate ticket for expanding typing to these places.